components/raftstore/src/store/fsm/peer.rs

```diff
@@ -4452,6 +4495,11 @@ where
                     "err" => %e,
                 );
             }
+        } else {
+            // It means that there does not exist pending uncompleted split (previous splits
+            // are already finished), so we can clear the previous uncompaigned
+            // list.
+            self.fsm.peer.uncampaigned_new_regions = None;
```

Comment:
Nit: I think it has already be cleared, becaues 1) if it's a leader this field will be cleared in `on_role_changed`, 2) if it's follower, this filed should be cleared on receive a message from leader.

Reply:
Yep, it's redundant here.